spec/services/govuk_emails/monitor_spec.rb

```diff
@@ -1,108 +1,181 @@
+        expect { call }.not_to have_enqueued_job
       end
     end
 
-    context "failure response" do
-      let(:status) { ScheduledMailing::FAILURE_STATUSES.sample }
-      let(:addressee) { scheduled_mailing.addressee }
-      let(:id) { scheduled_mailing.id }
-      let(:raven_message) { "Unable to deliver mail to #{addressee} - ScheduledMailing record #{id}" }
+    context "when on production host" do
+      before { allow(HostEnv).to receive(:production?).and_return(true) }
 
-      it "updates the scheduled_mailing record" do
-        subject
-        expect(scheduled_mailing.reload.status).to eq status
+      context "with a provider financial reminder with a \"delivered\" response from govuk notify" do
+        let(:scheduled_mailing) { create(:scheduled_mailing, :provider_financial_reminder, :processing) }
+        let(:status) { "delivered" }
+
+        it_behaves_like "non-failure handling job"
+
+        it "updates the scheduled_mailing record" do
+          expect { call }.to change { scheduled_mailing.reload.status }.from("processing").to("delivered")
+        end
       end
 
-      context "on production host" do
-        before { allow(HostEnv).to receive(:production?).and_return(true) }
+      context "with a provider financial reminder with a \"permanent-failure\" response from govuk notify" do
+        let(:scheduled_mailing) { create(:scheduled_mailing, :provider_financial_reminder, :processing) }
+
+        let(:status) { "permanent-failure" }
+        let(:addressee) { scheduled_mailing.addressee }
+        let(:id) { scheduled_mailing.id }
+
+        it "updates the scheduled_mailing record" do
+          expect { call }.to change { scheduled_mailing.reload.status }.from("processing").to("permanent-failure")
+        end
 
-        it "captures raven message" do
-          expect(AlertManager).to receive(:capture_message).with(raven_message)
-          subject
+        it "captures alert message" do
+          expect(AlertManager).to receive(:capture_message).with("Unable to deliver mail to #{addressee} - ScheduledMailing record #{id}")
+          call
         end
 
-        it "schedules an undeliverable alert email" do
-          expect { subject }.to change(ScheduledMailing, :count).by(1)
-          undeliverable_alert_mail = ScheduledMailing.order(:created_at).last
+        it "creates 1 scheduled_mailing to send alert email to the apply team", :aggregate_failures do
+          expect { call }.to change(ScheduledMailing, :count).by(1)
+
+          undelivered_mailing = ScheduledMailing.find_by(mailer_klass: "UndeliverableEmailAlertMailer")
+
+          expect(undelivered_mailing)
+            .to have_attributes(
+              legal_aid_application_id: scheduled_mailing.legal_aid_application_id,
+              mailer_klass: "UndeliverableEmailAlertMailer",
+              mailer_method: "notify_apply_team",
+              addressee: Rails.configuration.x.support_email_address,
```

Comment:
Is this config variable set via an environment variable?

I wonder if it is worth stubbing the config, or using something like [`climate_control`](https://github.com/thoughtbot/climate_control) so the test can be more specific and potentially less flaky?

Reply:
was in the original spec and does not seem to have problems.